connection.go

```diff
@@ -61,14 +61,15 @@ type redisConnection struct {
 }
 
 // OpenConnection opens and returns a new connection
-func OpenConnection(tag string, network string, address string, db int, errChan chan<- error) (Connection, error) {
-	redisClient := redis.NewClient(&redis.Options{Network: network, Addr: address, DB: db})
-	return OpenConnectionWithRedisClient(tag, redisClient, errChan)
+func OpenConnection(tag string, redisOption *redis.Options, errChan chan<- error) (Connection, error) {
+	// redisClient := redis.NewClient(&redis.Options{Network: network, Addr: address, DB: db})
+	redisClient := redis.NewClient(redisOption)
+	return OpenConnectionWithRmqRedisClient(tag, RedisSingleWrapper{redisClient}, errChan)
 }
 
-// OpenConnectionWithRedisClient opens and returns a new connection
-func OpenConnectionWithRedisClient(tag string, redisClient redis.Cmdable, errChan chan<- error) (Connection, error) {
-	return OpenConnectionWithRmqRedisClient(tag, RedisWrapper{redisClient}, errChan)
+// OpenConnectionWithRedisClusterClient opens and returns a new connection for a cluster redis
+func OpenConnectionWithRedisClusterClient(tag string, clusterClient *redis.ClusterClient, errChan chan<- error) (Connection, error) {
+	return OpenConnectionWithRmqRedisClient(tag, RedisClusterWrapper{clusterClient}, errChan)
```

Comment:
Have you tried using the existing wrapper with the cluster client?
```suggestion
	return OpenConnectionWithRmqRedisClient(tag, RedisSingleWrapper{clusterClient}, errChan)
```
When I compare these files
https://github.com/adjust/rmq/blob/3113601ad47836e110cd3201f160b69c172f9770/redis_wrapper_single.go
https://github.com/adjust/rmq/blob/3113601ad47836e110cd3201f160b69c172f9770/redis_wrapper_cluster.go
it looks like the cluster client has exactly the same functions, so I wonder if it might actually implement the `redis.Cmdable` interface, in which case we should be able to still use the existing wrapper instead of having the need for two different (but almost identical) wrapper implementations.

Reply:
Yes I have tried, works fine with me 💯

Reply:
> so I wonder if it might actually implement the redis.Cmdable interface

This also confuses me, the `redis.Client` does implement the `redis.Cmdable`, but, `redis.ClusterClient` does not.

I have checked the wrapper functions, which is common, and my team also wrapped same functions both for cluster and single redis.

It also needs some work to do to make the code more elegent, such as making new interface to fit the two type of redis clients, I will see what I can do in the future, thanks for pointing out! 👍

Reply:
The redis.ClusterClient has implementd Cmdable already, so this wrapper might be not necessary

Reply:
Thanks for jumping in here! 🙌 

Do you think you can take this branch and remove this wrapper again to see if it works without it? Feel free to open a new PR with the updated branch then. Thanks again!